### ISSUE TITLE
WebGPU CopyBufferToBuffer does not work as of version 3.1.65

### ISSUE THREAD

**Issue by klukaszek** [emscripten-core/emscripten#22699]
**Version of emscripten/emsdk:**
Version >= 3.1.65

**Failing command line in full:**
There's no problem with the command-line tool here. The issue is at runtime where any WebGPU calls to CopyBufferToBuffer() fail because the buffer ID for the destination buffer seems to not get properly passed to the function from C.

```
var _wgpuCommandEncoderCopyBufferToBuffer = (encoderId, srcId, srcOffset, dstId, dstoffset, size) => {
commandEncoder = WebGPU.mgrCommandEncoder.get(encoderId); 
var src= WebGPU.mgrBuffer.get(srcId); 
var dst = WebGPU.mgrBuffer.get(dstId);
commandEncoder.copyBufferToBuffer(src, srcOffset, dst, dstoffset, size);
};
```

This function seems to receive a dstId of 0 which results in WebGPU.mgrBuffer.get(dstId) failing. I have verified the value being passed through the C API and it does represent a valid buffer (and it still works in previous versions of Emscripten).

I originally found this issue because I work with an older version of Emscripten on my machine, but with my deployment, I was working with the latest version. I narrowed down the problem to Emscripten version 3.1.65+ where the CopyBufferToBuffer() function stops working.

I have discussed with @beaufortfrancois in DMs and he suggested looking into the following in case it was the breaking change:
https://github.com/emscripten-core/emscripten/pull/22301

I'm busy for the next week with school but I thought I would post the issue here in case anyone else is encountering the same problem.

**Testing**
- Ubuntu 22.04 (Local Machine)
- Ubuntu 22.04 (GitHub Runner)

**Comment by sbc100** [comment]
I think its likely a duplicate of #22682 which was fixed already in #22689?

**Comment by sbc100** [comment]
Actually the breaking that caused #22682  was not introduced until 3.1.67 so maybe different?

**Comment by klukaszek** [comment]
Seems to be similar where the produced JavaScript code is fine, but whatever is responsible for passing the buffer ID to the JS function isn't working properly.

**Comment by sbc100** [comment]
Can you try with `emsdk install tot` to see if this was fixed by #22689?

**Comment by sbc100** [comment]
I guess #22301 is more likely if you are sure that 3.1.65 is to blame.

@kainino0x what do you think is going on here?

**Comment by klukaszek** [comment]
> Can you try with `emsdk install tot` to see if this was fixed by #22689?

Yeah I can give that a try in a bit!

**Comment by klukaszek** [comment]
CopyBufferToBuffer() worked with TOT! I'll close the issue as it seems to have been resolved. Not sure if #22689 was responsible for the fix but it seems likely.

I'll stick with 3.1.64 until 3.1.69 is released!
